superset/viz.py

```diff
@@ -276,9 +276,14 @@ def cache_key(self, query_obj):
         for k in ['from_dttm', 'to_dttm']:
             del cache_dict[k]
 
-        for k in ['since', 'until', 'datasource']:
+        for k in ['since', 'until']:
             cache_dict[k] = self.form_data.get(k)
 
+        cache_dict['datasource'] = self.form_data.get(
```

Comment:
@mistercrunch I don't know if this should simply be `cache_dict['datasource'] = self.datasource.uid` instead.

Reply:
either should work, but `cache_dict['datasource'] = self.datasource.uid` is simpler